app/src/main/java/com/Found404/paypro/ui/pages/CardPaymentsPage.kt

```diff
@@ -93,22 +105,21 @@ fun CardPayments(
         )
         println("Card Types: $cardTypes")
 
-        Column(
-            modifier = Modifier
-                .fillMaxWidth()
-        ) {
-            for (cardType in cardTypes) {
+        Column(modifier = Modifier.fillMaxWidth()) {
+            cardTypes.forEach { cardType ->
+                var isChecked by remember { mutableStateOf(false) }
                 CreateRow(
-                    cardNameParam = cardType.name,
-                    cardTypeParam = merchantModel.cardTypes.contains(cardType.name)
-                ) {
-                    merchantModel = if (merchantModel.cardTypes.contains(cardType.name)) {
-                        merchantModel.copy(cardTypes = merchantModel.cardTypes - cardType.name)
+                    cardName = cardType.name,
+                    cardId = cardType.cardBrandId,
+                    isChecked = isChecked
+                ) { checked ->
+                    isChecked = checked
+                    merchantModel = if (isChecked) {
+                        merchantModel.copy(cardTypes = merchantModel.cardTypes + CardBrandType.values().first { it.cardBrandId == cardType.cardBrandId })
```

Comment:
I see that the hardcoded `CardBrandType` is still used here. Card brands received from the backend should only be used.  It is also not clear to me what this line does exactly